frontend/src/pages/NewRun.test.tsx

```diff
@@ -0,0 +1,1160 @@
+      await TestUtils.flushPromises();
+
+      expect(tree.state('pipeline')).toHaveProperty('parameters', originalRunPipelineParams);
+      tree.unmount();
+    });
+
+
+    it('shows a page error if getRun fails', async () => {
+      // Don't actually log to console.
+      // tslint:disable-next-line:no-console
+      console.error = jest.spyOn(console, 'error').mockImplementationOnce(() => null);
+
+      const runDetail = newMockRunDetail();
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.cloneFromRun}=${runDetail.run!.id}`;
+
+      TestUtils.makeErrorResponseOnce(getRunSpy, 'test error message');
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      await TestUtils.flushPromises();
+
+      expect(updateBannerSpy).toHaveBeenLastCalledWith(expect.objectContaining({
+        additionalInfo: 'test error message',
+        message: `Error: failed to retrieve original run: ${runDetail.run!.id}. Click Details for more information.`,
+        mode: 'error',
+      }));
+      tree.unmount();
+    });
+
+  });
+
+  describe('creating a new run', () => {
+
+    it('disables \'Create\' new run button by default', async () => {
+      const tree = shallow(<TestNewRun {...generateProps() as any} />);
+      await TestUtils.flushPromises();
+
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', true);
+      tree.unmount();
+    });
+
+    it('enables the \'Create\' new run button if pipeline ID in query params and run name entered', async () => {
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.pipelineId}=${newMockPipeline().id}`;
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'run name' } });
+      await TestUtils.flushPromises();
+
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', false);
+      tree.unmount();
+    });
+
+    it('re-disables the \'Create\' new run button if pipeline ID in query params and run name entered then cleared', async () => {
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.pipelineId}=${newMockPipeline().id}`;
+
+      const tree = shallow(<TestNewRun {...props as any} />);
+      (tree.instance() as any).handleChange('runName')({ target: { value: 'run name' } });
+      await TestUtils.flushPromises();
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', false);
+
+      (tree.instance() as any).handleChange('runName')({ target: { value: '' } });
+      expect(tree.find('#createNewRunBtn').props()).toHaveProperty('disabled', true);
+
+      tree.unmount();
+    });
+
+    it('sends a request to create a new run when \'Create\' is clicked', async () => {
+      const props = generateProps();
+      const experiment = newMockExperiment();
+      const pipeline = newMockPipeline();
+      props.location.search =
```

Comment:
nit: You can use `setState` in cases like this where you're not testing the interactions leading up to clicking the button to narrow the scope of the test.

Reply:
How strongly do you feel about this? I kind of prefer keeping the test as close as possible to the actual user flow, but I understand your point

Reply:
I think it's good to minimize what the test is testing, and if you have prior tests that are making sure the state is correct after certain interactions, then it's fine for this test to pick up from that point and assume what the state is.